pandas/core/indexes/base.py

```diff
@@ -3902,6 +3904,9 @@ def __getitem__(self, key):
         key = com.values_from_object(key)
         result = getitem(key)
         if not is_scalar(result):
+            if np.ndim(result) > 1:
```

Comment:
`deprecate_ndim_indexing` also checks `np.ndim(result) > 1`, so this check should be unnecessary.

Reply:
We have to do the check anyway, the only question is if we put this line inside or outside the check.  i prefer inside, but fine either way

Reply:
Ah, missed that we had the return here. This looks good.